lib/jsdom/living/nodes/HTMLSelectElement-impl.js

```diff
@@ -82,19 +83,69 @@ class HTMLSelectElementImpl extends HTMLElementImpl {
       selected.forEach((option, index) => {
         option._selectedness = index === selected.length - 1;
       });
+      this._selectedOption = selected[selected.length - 1];
     }
   }
 
+  _updateSelected(child) {
+    if (this.hasAttributeNS(null, "multiple")) {
+      return;
+    }
+    if (child.hasAttributeNS(null, "disabled")) {
+      return;
+    }
+    const optArr = this._toOptionArray(child);
+    if (!optArr.length) {
+      return;
+    }
+    const selectedOptions = optArr.filter(opt => opt._selectedness);
+    // If any selected is last selected
+    if (selectedOptions.length) {
+      // pushed at least one element with selected=true, reset
+      this._askedForAReset();
+      return;
+    }
+
+    // No selection so first option is selected
+    if ((!this._selectedOption || !this._selectedOption._selectedness) && this._displaySize === 1) {
+      this._selectedOption = optArr[0];
+      this._selectedOption._selectedness = true;
+    }
+  }
+
+  _toOptionArray(child) {
+    const array = [];
+    if (child._localName === "option") {
+      array.push(child);
+    } else if (child._localName === "optgroup") {
+      for (const childOfGroup of domSymbolTree.childrenIterator(child)) {
+        if (childOfGroup._localName === "option" && !childOfGroup.hasAttributeNS(null, "disabled")) {
+          array.push(childOfGroup);
+        }
+      }
+    }
+    return array;
+  }
+
   _descendantAdded(parent, child) {
     if (child.nodeType === NODE_TYPE.ELEMENT_NODE) {
-      this._askedForAReset();
+      if (parent === this) {
+        this._updateSelected(child);
+      } else if (parent._localName === "optgroup") {
+        this._updateSelected(parent);
+      } else {
+        this._askedForAReset();
```

Comment:
It's a bit weird to call `_askedForAReset()` when nothing actually asked for a form reset. Let's rename `_askedForAReset()` to `_selectednessSettingAlgorithm()` (see https://html.spec.whatwg.org/#selectedness-setting-algorithm), then add a quick wrapper `_askedForAReset() { this._selectednessSettingAlgortihm(); }`.

After you've done that, maybe it'd make sense to rename `_updateSelected()` to something related, like `_selectednessSettingAlgorithmWithKnownNewAddition(newChild)` or `_maybeRunSelectednessSettingAlgorithm(newChild)` or something? To make it clear it's an optimization related to minimizing how often we run the full selectedness setting algorithm.

Reply:
Done the rename and wrapper and the new name for the fast path method.